src/h3/api/_api_template.py

```diff
@@ -172,32 +267,50 @@ def hex_ranges(hexes, K):
         return out
 
     def k_ring_distances(h, K):
+        """ Alias for `hex_range_distances`.
+        """
         return hex_range_distances(h, K)
 
     def h3_to_children(h, res=None):
         """ Get the children of a hexagon.
+
         Parameters
         ----------
         h : H3 address
         res : int or None, optional
             The resolution for the children.
             If `None`, then `res = resolution(h) + 1`
+
         Returns
         -------
-        collection of h3 addresses
+        UnorderedCollection[H3Cells]
         """
         mv = _cy.children(_in_scalar(h), res)
 
         return _out_unordered(mv)
 
     # todo: nogil for expensive C operation?
     def compact(hexes):
+        """ Compact a collection of H3 Cells by combining
+        smaller cells into larger cells, if all child cells
+        are present
+
+        Returns
+        -------
+        UnorderedCollection[H3 Cells]
+
+        """
         hu = _in_collection(hexes)
         hc = _cy.compact(hu)
 
         return _out_unordered(hc)
 
     def uncompact(hexes, res):
+        """ Reverse the `compact` operation and return a collection
+        of H3 Cells, all of resolution `res`.
+
+        what if uncompact input contains a hex samller than res?
```

Comment:
An error code will be returned: https://github.com/uber/h3/blob/master/src/h3lib/lib/h3Index.c#L425

Reply:
got it, thanks

Reply:
I'll add a test for this.